src/backend/distributed/planner/multi_physical_planner.c

```diff
@@ -2091,12 +2091,21 @@ SubquerySqlTaskList(Job *job, PlannerRestrictionContext *plannerRestrictionConte
 	 * That means all tables are reference tables and we can pick any any of them
 	 * as an anchor table.
 	 */
-	if (targetCacheEntry == NULL)
+	if (targetCacheEntry == NULL && list_length(rangeTableList) >= 1)
 	{
 		RangeTblEntry *rangeTableEntry = (RangeTblEntry *) linitial(rangeTableList);
 		relationId = rangeTableEntry->relid;
 		targetCacheEntry = DistributedTableCacheEntry(relationId);
 	}
+	else if (targetCacheEntry == NULL && list_length(rangeTableList) < 1)
```

Comment:
This looks slightly odd. The `list_length(rangeTableList) < 1` is redundant. 

but maybe just:

```
if (targetCacheEntry == NULL)
{
    RangeTblEntry *rangeTableEntry = NULL;

    if (list_length(rangeTableList) == 0)
    {
       // error
    }

    ...
}
```